tests/e2e/__testers__.ts

```diff
@@ -92,60 +130,61 @@ export const testTemplate = (params: {
     /**
      * Drop database before running the tests case it wasn't cleaned up from the previous test run.
      */
+
     await ctx.dropTestDatabase()
+    await ctx.initTestDatabase()
+
+    const initResult = await ctx.runAsync(`npm run init`, { reject: true })
 
-    /**
-     * Test 1
-     * Check that the initialization script works. This includes running migrate triggering generators and executing the seed.
-     */
-    const initResult = ctx.run(`npm run init`, { reject: true })
     expect(initResult.stderr).toMatch('')
     expect(stripAnsi(initResult.stdout)).toMatch('Generated Prisma Client')
     expect(stripAnsi(initResult.stdout)).toMatch('Running seed command')
 
+    // TODO Test the Vercel API (next dev for Blog template)
+    // await ctx.fs.writeAsync('.vercel/project.json', {
+    //   projectId: 'prj_6yrTe9CGQagAQwGjr7JEejkxhz3A',
+    //   orgId: 'team_ASKXQ5Yc1an2RqJc5BCI9rGw',
+    // })
+  })
+
+  /**
+   * Test 2
+   * Check that the template migration script works.
+   */
+  it(`${params.templateName} - template migration script should work`, async () => {
+    if (ctx.template._tag === 'Empty') return
```

Comment:
This if block should surround the test. Right now it looks like the empty template passes this test, which is misleading.

Reply:
Funny to see how trivial thing like not wanting to run a test if the tag is empty breaks the whole test suit.
Basically wrapping around the `it` with the `if (ctx.template._tag === 'Empty')` throws an error because `tag` i not defined yet. (that is also why breaking tests into smaller bits is useful)

The template is defined in `beforeAll`. Will see if there is a quick way around this.

Reply:
It makes sense, you're accessing `ctx` which is built up in the lifecycle hooks. You need to work with data available before the test runs. Use the parameters passed to `testTemplate`.